src/App.tsx

```diff
@@ -1,16 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+import { Person } from './types/Person';
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
```

Comment:
This `debounce` function will not work as expected. Every time `handleQuery` is called, a new debounced function is created. To fix this, you should create `debouncedSetAppliedQuery` outside of the `handleQuery` function and call it inside `handleQuery`.